src/jaxsim/mujoco/loaders.py

```diff
@@ -0,0 +1,480 @@
+            if geometry_element.attrib.get("name") in collision_names:
+                r, g, b, a = geometry_element.attrib["rgba"].split(" ")
+                geometry_element.set("rgba", f"{r} {g} {b} 0")
+
+        # -----------------------
+        # Create the scene assets
+        # -----------------------
+
+        asset_element = (
+            ET.SubElement(mujoco_element, "asset")
+            if len(mujoco_element.findall(".//asset")) == 0
+            else mujoco_element.find(".//asset")
+        )
+
+        _ = ET.SubElement(
+            asset_element,
+            "texture",
+            type="skybox",
+            builtin="gradient",
+            rgb1="0.3 0.5 0.7",
+            rgb2="0 0 0",
+            width="512",
+            height="512",
+        )
+
+        _ = ET.SubElement(
+            asset_element,
+            "texture",
+            name="plane_texture",
+            type="2d",
+            builtin="checker",
+            rgb1="0.1 0.2 0.3",
+            rgb2="0.2 0.3 0.4",
+            width="512",
+            height="512",
+            mark="cross",
+            markrgb=".8 .8 .8",
+        )
+
+        _ = ET.SubElement(
+            asset_element,
+            "material",
+            name="plane_material",
+            texture="plane_texture",
+            reflectance="0.2",
+            texrepeat="5 5",
+            texuniform="true",
+        )
+
+        # ----------------------------------
+        # Populate the scene with the assets
+        # ----------------------------------
+
+        worldbody_scene_element = ET.SubElement(mujoco_element, "worldbody")
+
+        _ = ET.SubElement(
+            worldbody_scene_element,
+            "geom",
+            name="floor",
+            type="plane",
+            size="0 0 0.05",
+            material="plane_material",
+            condim="3",
+            contype="1",
+            conaffinity="1",
+        )
+
+        _ = ET.SubElement(
+            worldbody_scene_element,
+            "light",
+            name="sun",
+            mode="fixed",
+            directional="true",
+            castshadow="true",
+            pos="0 0 10",
+            dir="0 0 -1",
+        )
+
+        # ------------------------------------------------
+        # Add a light following the  CoM of the first link
+        # ------------------------------------------------
+
+        if not rod_model.is_fixed_base():
+
+            worldbody_element = None
+
+            # Find the <worldbody> element of our model by searching the one that contains
+            # all the considered joints. This is needed because there might be multiple
+            # <worldbody> elements inside <mujoco>.
+            for wb in mujoco_element.findall(".//worldbody"):
+                if all(
+                    [
+                        wb.find(f".//joint[@name='{j}']") is not None
+                        for j in considered_joints
+                    ]
+                ):
```

Comment:
```suggestion
                if all(
                        wb.find(f".//joint[@name='{j}']") is not None
                        for j in considered_joints
                ):
```
You could use a generator to avoid unnecessary memory usage

Reply:
Addressed in https://github.com/ami-iit/jaxsim/pull/83/commits/4dcd4a760ac2d9afe7d6a74fea385f2447ebb07f